com.unity.ml-agents/Runtime/SideChannels/EngineConfigurationChannel.cs

```diff
@@ -13,7 +13,7 @@ public class EngineConfigurationChannel : SideChannel
         /// <summary>
         /// Initializes the side channel.
         /// </summary>
-        public EngineConfigurationChannel()
+        internal EngineConfigurationChannel()
```

Comment:
Since this is basically a singleton, I don't think it should have a public constructor.

Reply:
Might be worth adding that as a comment to this constructor and the one for stats side channel.